evm/src/cpu/cpu_stark.rs

```diff
@@ -60,11 +60,14 @@ fn ctl_data_binops<F: Field>(ops: &[usize]) -> Vec<Column<F>> {
 
 /// Create the vector of Columns corresponding to the three inputs and
 /// one output of a ternary operation.
-fn ctl_data_ternops<F: Field>(ops: &[usize]) -> Vec<Column<F>> {
+/// If `is_shift` is `true`, we offset the memory_channels indices used for the inputs
+/// by 1. It will only be valid if the associated filter is one of the two shift flags.
+fn ctl_data_ternops<F: Field>(ops: &[usize], is_shift: bool) -> Vec<Column<F>> {
```

Comment:
Could you explain the need for `is_shift` in a bit more clearly please (both here and in the comment)? I assume it's implementing the mapping described at https://github.com/mir-protocol/plonky2/blob/d51a512caae653e17b2591c988f75414e395e8f8/evm/src/cpu/shift.rs#L53 but it would be helpful to explicitly document which memory channel corresponds to which value when `offset = 1` (including the precomputed value).

Reply:
We need this offset because the shift operations use the first memory channel to store the shift, which is not used when computing the related arithmetic operation.
Because both MUL/DIV are binary ops, we can just switch all 3 channels used for ternary operations and safely ignore the last one.
I'll update the documentation to make it clearer!